HeliPort/NetworkManager.swift

```diff
@@ -186,23 +186,16 @@ final class NetworkManager {
     }
 
     class func getRouterAddress(bsd: String) -> String? {
-        let dynamicCreate = SCDynamicStoreCreate(kCFAllocatorDefault, "router-ip" as CFString, nil, nil)
-        let keyIPv4 = "State:/Network/Global/IPv4" as CFString
-        let keyIPv6 = "State:/Network/Global/IPv6" as CFString
-        let dictionary = SCDynamicStoreCopyValue(dynamicCreate, keyIPv4)
-            ?? SCDynamicStoreCopyValue(dynamicCreate, keyIPv6)
-
-        guard let interface = dictionary?[kSCDynamicStorePropNetPrimaryInterface] as? String, interface == bsd else {
-            Log.error("Could not find interface")
+        let routerCommand = ["netstat -rn | egrep -o default.*\(bsd)"]
+        guard let routerOutput = commandLine(args: routerCommand) else {
             return nil
         }
+        let ipv4Command = [#"echo "\#(routerOutput)" | egrep -o "\#(ipv4Pattern)""#]
+        let ipv4Output = commandLine(args: ipv4Command)
```

Comment:
You don't need to run a shell command to extract the address. Try extracting it with `NSRegularExpression` directly.

Reply:
Got it